File: langs/javascript/5-2.js
const run = require("./utils/run");

module.exports = (input) => {
  let has_pair = /.*(\w\w).*\1/;
  let has_middle = /.*(\w).\1/;

  return has_pair.test(input) && has_middle.test(input);
};

run(module.exports);

File: langs/rust/src/bin/5-2.rs
use rust::{check, run};

fn solution(input: &str) -> usize {
    input.lines().map(is_good).filter(|b| *b).count()
}

fn is_good(s: &str) -> bool {
    let has_pair = r".*(\w\w).*\1";
    let has_middle = r".*(\w).\1";
    check(has_pair, s) && check(has_middle, s)
}

fn main() {
    run(solution);
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_5_1() {
        assert_eq!(is_good("qjhvhtzxzqqjkmpb"), true);
        assert_eq!(is_good("xxyxx"), true);
        assert_eq!(is_good("uurcxstgmygtbstg"), false);
        assert_eq!(is_good("ieodomkazucvgmuy"), false);
    }
}
